src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -154,6 +154,25 @@ export class IndexPatternsService {
     }));
   };
 
+  /**
+   * Returns an object matching a given title
+   *
+   * @param title {string}
+   * @returns {Promise<SavedObject|undefined>}
+   */
+  findByTitle = async (title: string, refresh: boolean = false) => {
+    if (!this.savedObjectsCache || refresh) {
```

Comment:
I think this could rely on `getIdsWithTitle`, filter based on the result, and then `get`

Lets call this `getByTitle`